### ISSUE TITLE
Reserve a seriesId from the CN when a new ProjectModel is created

### ISSUE THREAD

**Issue by laurenwalker** [NCEAS/metacatui#988]
When a new ProjectModel is created, a seriesId is generated and reserved in the CN through [CN.reserveIdentifier()](https://releases.dataone.org/online/api-documentation-v2.0/apis/CN_APIs.html#CNCore.reserveIdentifier)

The seriesId needs to be reserved since it will be used in the `isPartOf` filter in the collection definition. #989

**Comment by laurenwalker** [comment]
Is there a way we can render the project editor without the series id, and send the call to reserve it while the view is rendering? That way at least something is happening on screen while the call is being completed. Especially on slower connections, that delay may be apparent.

**Comment by robyngit** [comment]
The only issue is that the `DataCatalogViewWithFilters` needs filters to render, and the Data tab is by default the first section a user views in a new project. I'll try starting the view with empty/default `Filters` then updating.

**Comment by laurenwalker** [comment]
Ah, ok. We could have the `EditCollectionView` `listenTo` the `seriesId` change rather than the `ProjectEditorView`, when it determines that there is no seriesId AND ProjectModel.isNew() == true. At least then, the rest of the editor will be rendered and ready to go.

Also, what do you think about defaulting to the Settings tab for new projects? Since that is where the URL short name is set? I could see either way working well.

**Comment by robyngit** [comment]
I just added a commit that makes the `DataCatalogViewWithFilters` render with empty filters, but now the search results change from showing every dataset to zero datasets once the SID is reserved. I think adding a listener to `EditCollectionView` is a better idea.

I also agree that showing the Settings tab first makes sense if the project is completely empty. I think ideally, it would be nice to eventually have a simple interface for users to enter the very basic project information (like a label and title) before getting to the project editor.

**Comment by laurenwalker** [comment]
Sorry, one more thing about this ticket - I just saw the code for calling the `generate` API. We actually want to use the `reserve` API instead, so that the identifier can't be used by other objects.

We should also try to send all XHRs in models, and not views. So the ProjectModel should have a `reserveSeriesId()` function that makes this call, instead of sending the XHR directly in the ProjectEditorView. 

That way the call can be used from anywhere in the app and it works exactly the same. It also keeps interaction with the server side isolated to models, while the views handle the interaction between the client-side code and the user interface.

**Comment by robyngit** [comment]
Oh thanks, that's good to know! I'll move that function to the model. Does it make more sense in the project model or the collection model (i.e. would we ever have a need to reserve an ID for a collection)?

About reserving the ID: Although it's not explicitly stated anywhere in the DataOne API docs (as far as I can tell), I believe that `generate` also reserves the identifier. Initially I called `generate` and then `reserve`, but `reserve` gave an error saying that the identifier was already reserved by me (user with my ORCID).

**Comment by laurenwalker** [comment]
Ah, good point, let's put this in the CollectionModel.

`generate` may `reserve` an id, I'm not actually sure about that. But let's use `reserve` so that we aren't waiting on the response from `generate` to get the seriesId value. It may also be useful to keep the control of the `seriesId` creation in MetacatUI, in case we ever want to support different id types/versions of UUID, etc.

BTW, I tried the `reserve` API and I got an error... I found out that the DataONE API is incorrect and the `id` data parameter sent with the request should actually be `pid`.
